tealish/nodes.py

```diff
@@ -10,26 +20,42 @@
 LITERAL_BYTES = r'"(.+)"'
 VARIABLE_NAME = r"[a-z_][a-zA-Z0-9_]*"
 
+if TYPE_CHECKING:
+    from . import TealishCompiler, TealWriter
+
 
 class Node(BaseNode):
-    pattern = ""
-    possible_child_nodes = []
+    pattern: str = ""
+    possible_child_nodes: List[Type[BaseNode]] = []
+
+    def __init__(
+        self,
+        line: str,
+        parent: Optional["Node"] = None,
+        compiler: Optional["TealishCompiler"] = None,
```

Comment:
I'd like to make these non optional if possible but will need to dig deeper on where/how its called